transport.go

```diff
@@ -0,0 +1,123 @@
+package lumigotracer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Transport struct {
+	rt         http.RoundTripper
+	provider   trace.TracerProvider
+	propagator propagation.TextMapPropagator
+}
+
+func NewTransport(transport http.RoundTripper) *Transport {
+	return &Transport{
+		rt:         transport,
+		provider:   otel.GetTracerProvider(),
+		propagator: otel.GetTextMapPropagator(),
+	}
+}
+
+func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	traceCtx, span := t.provider.Tracer("lumigo").Start(req.Context(), "HttpSpan")
+
+	req = req.WithContext(traceCtx)
+	span.SetAttributes(semconv.HTTPClientAttributesFromHTTPRequest(req)...)
+	span.SetAttributes(semconv.HTTPTargetKey.String(req.URL.Path))
+	span.SetAttributes(semconv.HTTPHostKey.String(req.URL.Host))
+	t.propagator.Inject(traceCtx, propagation.HeaderCarrier(req.Header))
+
+	if req.Body != nil {
+		bodyBytes, bodyErr := io.ReadAll(req.Body)
+		if bodyErr != nil {
+			return
```

Comment:
If there was a problem, maybe we should continue either way, in order to grab as much data as possible (e.g. headers)?

Reply:
also in line 76